example/passthrough_hp.cc

```diff
@@ -946,8 +974,30 @@ static void do_write_buf(fuse_req_t req, size_t size, off_t off,
         FUSE_BUF_IS_FD | FUSE_BUF_FD_SEEK);
     out_buf.buf[0].fd = fi->fh;
     out_buf.buf[0].pos = off;
+    fuse_buf_copy_flags copy_flags = FUSE_BUF_COPY_FLAGS;
+
+    const int bypass = fs.bypass_rw;
+
+    if (bypass) {
+
+        /* special benchmark, to see how fast fuse works - the destination
+         * file is not modified
+         */
+        if (bypass == SFS_BYPASS_BUF_COPY) {
+            /* the received buffer is not used at all */
+            size_t res = size;
+            fuse_reply_write(req, (size_t)res);
+            return;
+        }
 
-    auto res = fuse_buf_copy(&out_buf, in_buf, FUSE_BUF_COPY_FLAGS);
+        /* the received buffer is used for another memory copy */
+        out_buf.buf[0].fd = -1;
```

Comment:
Instead of overwriting the previous definiton of out_buf, can we move the first definition into an `else` clause?